frontend/src/components/App/TopBar.tsx

```diff
@@ -71,6 +71,10 @@ export default function TopBar({}: TopBarProps) {
   const history = useHistory();
   const { appBarActions, appBarActionsProcessors } = useAppBarActionsProcessed();
 
+  // getToken may return stale value so we need to rerender on navigation
+  // TODO: create a useToken hook that always returns latest token
+  // eslint-disable-next-line no-unused-vars
+  const _location = useLocation();
```

Comment:
Why not make the hasToken a const set by a useMemo, with the cluster being a dependency? Would that fix this?

Reply:
Got it. Thanks for elaborating.